tools/autograd/gen_python_functions.py

```diff
@@ -113,6 +115,22 @@
 PY_VARIABLE_METHOD_DEF = CodeTemplate("""\
 {"${name}", (PyCFunction)${pycname}, ${flags}, NULL},""")
 
+PY_RETURN_NAMEDTUPLE_DEF = CodeTemplate("""\
+static PyStructSequence_Field fields${namedtuple_type_index}[] = {
+  ${namedtuple_fields} {nullptr}
```

Comment:
I think the overread problem is due to how `PyStructSequence_Field` is defined, although I'm not sure how this ever passed the tests in the first place. You have expression `{"foo", ""}` in a situation where it should be a pointer.

Reply:
@ezyang I'm not sure if I get your point, the way currently it is done generate codes like:

```C++
static PyStructSequence_Field fields0[] = {
    {"U", ""}, {"S", ""}, {"V", ""}, {nullptr}
};
```
The way this was done was learned from https://github.com/python/cpython/blob/621cebe81b1e6c8de10425955bf532d31ee4df42/Python/thread.c#L152

```C
static PyStructSequence_Field threadinfo_fields[] = {
    {"name",    "name of the thread implementation"},
    {"lock",    "name of the lock implementation"},
    {"version", "name and version of the thread library"},
    {0}
};
```
Is there anything wrong?

Reply:
I guess not! TIL.